### PR TITLE
[Core, HTML2] Rendered docs are incomplete

### PR DESCRIPTION
**Pull request by richardwhiuk**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.
  - No technical committee for HTML 2.

### Description of the PR

- Simple aliased types are now passed to the generator

- Body parameters are now provided with their schemas

- Schema block is now visible in the generated documentation

Note, this changes some of the core behaviour, which doesn't pass aliased models to the generator, so may have large scale impacts.

Changes are by @MalachiJones, @mthebridge, @richardwhiuk and others at @Metaswitch 

This fixes https://github.com/OpenAPITools/openapi-generator/issues/1441

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/htmlDocs2/paramB.mustache [hunk 1]
```diff
@@ -4,7 +4,11 @@
 <script>
 $(document).ready(function() {
   var schemaWrapper = {{{jsonSchema}}};
-  var schema = schemaWrapper.schema;
+
+  var schema = findNode('schema',schemaWrapper).schema;
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/htmlDocs2/paramB.mustache]: will find node alway return a result here?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java [hunk 2]
```diff
@@ -488,12 +488,12 @@ private Model getParent(Model model) {
                 // TODO revise below as we've already performed unaliasing so that the isAlias check may be removed
                 Map<String, Object> modelTemplate = (Map<String, Object>) ((List<Object>) models.get("models")).get(0);
                 // Special handling of aliases only applies to Java
-                if (modelTemplate != null && modelTemplate.containsKey("model")) {
-                    CodegenModel m = (CodegenModel) modelTemplate.get("model");
-                    if (m.isAlias) {
-                        continue;  // Don't create user-defined classes for aliases
-                    }
-                }
+                // if (modelTemplate != null && modelTemplate.containsKey("model")) {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java]: can you explain what is the impact of this change?

**Comment by macjohnny** [comment]
@richardwhiuk thanks for your PR!

**Comment by eclipse1985** [comment]
Is this related to https://github.com/OpenAPITools/openapi-generator/issues/2427 ?

**Comment by frank-schoenheit-red6es** [comment]
anybody here who can bring this forward? Recently migrated to openapi-generator from swagger-generator, and using the html generator as an opportunity to publish a "have a quick look" version of the API documentation during a Jenkins build. The bug here breaks this approach :-(

**Comment by macjohnny** [comment]
@richardwhiuk can you resolve the conflicts?

**Comment by richardwhiuk** [comment]
@macjohnny Yeah, I'll resolve the conflicts.

We've spotted some bugs that occur with this fix that I think should be covered along with this fix - I'll try and get those resolved this week.

**Comment by frank-schoenheit-red6es** [comment]
thanks for your efforts, guys!

**Comment by eclipse1985** [comment]
@richardwhiuk any update :)?

**Comment by Fjolnir-Dvorak** [comment]
Thank you for the merge request. I tried it a few minutes ago and merged it with the latest master. Simple objects are working but if objects are nested it still does not work.
I hope I will remember to design a simple test yaml to visualize the error which I will post here

**Comment by rachanadeshmukh** [comment]
Hi, are there any updates on the status of merging this fix? My team is currently dependent on this in order to generate static html2 documentation based on openapi specs that contain all the required annotated documentation (including request schemas). We are using the gradle plugin for this.

**Comment by JonarJr** [comment]
any updates for this fix?

**Comment by masci** [comment]
Patch in https://github.com/OpenAPITools/openapi-generator/pull/5643 works for me

**Comment by jimschubert** [comment]
I've merged master into this code and opened/merged #7510 

As mentioned earlier, #5643 was opened with a similar fix, but after checking the code between the two branches it looks like that PR just copied this code and refactored. I'm pretty big on attribution of contributions, so I've replaced the `findNode` changes from #5643 with the `findNode` implementation from this PR.

**Pull request closed without merging**